prime-router/docs/proposals/0009-proposal-for-history-api.md

```diff
@@ -0,0 +1,232 @@
+
+Example Json:
+```json
+{
+    "result_set": {
+        "count": 20,
+        "offset": 0,
+        "limit": 20,
+        "total": 3
+    },
+    "submissions": [
+        {
+            "actionId": 143,
+            "reportId" : "18a8e461-b3f8-436d-b94e-4065c7b9d186",
+            "createdAt": "2021-06-30 13:03:00.305657-04",
+            "reportItemCount" : 23,
+            "destinationCount" : 5,
+            "warningCount" : 8,
+            "errorCount" : 0,
+            "status": "CREATED"
+        },
+        {
+            "actionId": 158,
+            "reportId" : null,
+            "createdAt": "2021-06-30 13:03:00.305657-04",
+            "reportItemCount" : 0,
+            "destinationCount" : 0,
+            "warningCount" : 5,
+            "errorCount" : 3,
+            "status": "BAD_REQUEST"
+        },
+        {
+            "actionId": 177,
+            "reportId" : null,
+            "createdAt": "2021-06-30 13:03:00.305657-04",
+            "reportItemCount" : 0,
+            "destinationCount" : 0,
+            "warningCount" : 0,
+            "errorCount" : 0,
+            "status": "EXCEPTION"
+        }
+    ]
+}
+```
+To simplify, the status could be made binary (pass/fail) with more detail added on a separate API call. Also, adding a new HttpStatus column to the action table for the receive results would simplify having to parse action_result to determine if the POST was successful. Alternatively, we could assume that if no report_file record was created for the action then the POST failed. However, I think the HttpStatus would be more useful long term.
+
+### Action Details for a Given Id
+`GET /reports/{organization}/action/{id}`
+
+As a followup request to the above endpoint, this provides more details on the specific action referenced in the results above. The response on this endpoint will show the metadata stored for the specific action. This shold be limited to receive actions only and check the action_params to ensure the organizations match.
+
+Example Json:
+```json
+{
+    "actionId": 233,
+    "actionName": "receive",
+    "createdAt": "2021-06-30 13:03:00.305657-04",
+    "status": "BAD_REQUEST",
+    "actionParams": "{\r\n   \"method\" : \"POST\",\r\n   \"Headers\" : {\r\n     \"connection\" : \"keep-alive\",\r\n     \"content-type\" : \"text\/csv\",\r\n     \"accept\" : \"text\/html, image\/gif, image\/jpeg, *; q=.2, *\/*; q=.2\",\r\n     \"host\" : \"localhost:7071\",\r\n     \"user-agent\" : \"Java\/11.0.10\",\r\n     \"content-length\" : \"0\",\r\n     \"client\" : \"ignore.ignore-simple-report\"\r\n   },\r\n   \"QueryParameters\" : { }\r\n }",
```

Comment:
maybe leave this out?

Oh, and while I'm thinking of it -- we'll need to add sending_org and sending_org_client as columns on the action table - otherwise we can't secure/auth on failed submissions that have no report_id!